extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -474,67 +473,6 @@ void validateConfigPropertiesInjectionPoints(
-            ValidationPhaseBuildItem validationPhase,
-            List<ConfigClassBuildItem> configClasses,
-            List<ConfigInjectionStaticInitBuildItem> configInjectionStaticInit,
-            BuildProducer<RunTimeConfigurationDefaultBuildItem> runTimeConfigurationDefault) {
-
-        // Add here annotated classes that are initialized during static init
-        Set<DotName> declaringClassCandidates = configInjectionStaticInit.stream()
-                .map(ConfigInjectionStaticInitBuildItem::getDeclaringCandidate).collect(toSet());
-
-        Set<Type> configClassesTypes = configClasses.stream().map(ConfigClassBuildItem::getTypes).flatMap(Collection::stream)
-                .collect(toSet());
-
-        for (InjectionPointInfo injectionPoint : validationPhase.getContext().getInjectionPoints()) {
-            if (injectionPoint.getType().name().equals(DotNames.INSTANCE)) {
-                continue;
-            }
-
-            Type type = Type.create(injectionPoint.getRequiredType().name(), Type.Kind.CLASS);
-            DotName injectionTypeName = null;
-            if (type.name().equals(MP_CONFIG) || type.name().equals(SR_CONFIG)) {
-                injectionTypeName = type.name();
-            } else if (injectionPoint.getRequiredQualifier(MP_CONFIG_PROPERTY_NAME) != null) {
-                injectionTypeName = MP_CONFIG_PROPERTY_NAME;
-            } else if (configClassesTypes.contains(type)) {
-                injectionTypeName = type.name();
-            }
-
-            if (injectionTypeName != null) {
-                AnnotationTarget target = injectionPoint.getTarget();
-                if (FIELD.equals(target.kind())) {
-                    FieldInfo field = target.asField();
-                    ClassInfo declaringClass = field.declaringClass();
-                    Map<DotName, List<AnnotationInstance>> annotationsMap = declaringClass.annotationsMap();
-                    for (DotName declaringClassCandidate : declaringClassCandidates) {
-                        List<AnnotationInstance> annotationInstances = annotationsMap.get(declaringClassCandidate);
-                        if (annotationInstances != null && annotationInstances.size() == 1) {
-                            AnnotationInstance annotationInstance = annotationInstances.get(0);
-                            if (annotationInstance.target().equals(declaringClass)) {
-                                LOGGER.warn("Directly injecting a " +
-                                        injectionTypeName +
-                                        " into a " +
-                                        declaringClassCandidate +
-                                        " may lead to unexpected results. To ensure proper results, please " +
-                                        "change the type of the field to " +
-                                        ParameterizedType.create(DotNames.INSTANCE, new Type[] { type }, null) +
```

Comment:
I see this earlier version of the check took into account the type being injected in the error message (i.e. it would suggest `Instance<MyConfig>` instead of `Instance<String>`, IIUC).

Would it make sense to still do that with the new approach, or is it too complex to be worth it?